code/game/mecha/equipment/weapons/weapons.dm

```diff
@@ -101,6 +101,17 @@
 	projectile = /obj/item/projectile/beam/pulse/heavy
 	fire_sound = 'sound/weapons/marauder.ogg'
 
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/plasma

```

Comment:
This cannot be attached to a Ripley, as they do not accept weapons. Remember to make an exception for this, or it will only work on combat mechs.